crates/blockchain/mempool.rs

```diff
@@ -100,8 +96,8 @@ pub fn filter_transactions(
 }
 
 /// Remove a transaction from the mempool
-pub fn remove_transaction(hash: H256, store: &Store) -> Result<(), StoreError> {
-    store.remove_transaction_from_pool(hash)
+pub fn remove_transaction(address: Address, hash: H256, store: &Store) -> Result<(), StoreError> {
+    store.remove_transaction_from_pool(address, hash)
```

Comment:
if you're passing the hash, why do you need the address?

Reply:
The mempool is now a `Hash<Address, MempoolTransactionStore>`, to either add or remove a transaction to it we need the sender's address for the first indexing there.